### PR TITLE
[Improve][connector-v2][HdfsFileSink]Delete temporary files not caused by checkpoint.

### PR DESCRIPTION
**Pull request by lightzhao**
When the checkpoint is not completed, if the flink job is killed or canceled, dirty files will be left under tmp and should be deleted.

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by lightzhao** [comment]
@EricJoy2048 @Hisoka-X PTAL

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/ParquetWriteStrategy.java [hunk 1]
```diff
@@ -299,4 +299,8 @@ private Schema buildAvroSchemaWithRowType(SeaTunnelRowType seaTunnelRowType, Lis
         MessageType seaTunnelRow = Types.buildMessage().addFields(types.toArray(new Type[0])).named("SeaTunnelRecord");
         return schemaConverter.convert(seaTunnelRow);
     }
+
+    @Override
+    public void close() {
+    }
```
Comment (liugddx) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/ParquetWriteStrategy.java]: There's no operation here?

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/OrcWriteStrategy.java [hunk 2]
```diff
@@ -359,4 +359,8 @@ private void setDoubleVector(Object value, DoubleColumnVector doubleVector, int
             throw new FileConnectorException(CommonErrorCode.ILLEGAL_ARGUMENT, errorMsg);
         }
     }
+
+    @Override
+    public void close() {
+    }
```
Comment (liugddx) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/OrcWriteStrategy.java]: There's no operation here?

**Comment by liugddx** [comment]
Can you provide test cases?

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/ParquetWriteStrategy.java (hunk 1 above)
Comment (lightzhao) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/ParquetWriteStrategy.java]: update complete.

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/OrcWriteStrategy.java (hunk 2 above)
Comment (lightzhao) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/OrcWriteStrategy.java]: update complete.

**Comment by lightzhao** [comment]
> Can you provide test cases?

I have completed the test. When the checkpoint is not completed, I kill the flink job. The files in the temporary directory are deleted successfully.

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/AbstractWriteStrategy.java [hunk 3]
```diff
@@ -328,6 +329,24 @@ public String getTargetLocation(@NonNull String seaTunnelFilePath) {
...
+     * Delete temporary files not caused by checkpoint.
+     * @param tmpFilePaths temporary files.
+     * @return void
+     */
+    public void deleteTmpFileWithNoCheckPoint(Set<String> tmpFilePaths){
+        try {
+            if (tmpFilePaths != null && tmpFilePaths.size() > 0) {
+                for (String filePath : tmpFilePaths){
+                    fileSystemUtils.deleteFile(filePath);
+                    log.info("delete no commit file:{}", filePath);
```
Comment (FlechazoW) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/AbstractWriteStrategy.java]: What's the meaning of 'no commit file'? Why not use the 'temporary file ' or 'uncommitted file'?

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/AbstractWriteStrategy.java [hunk 4]
```diff
@@ -328,6 +329,24 @@ public String getTargetLocation(@NonNull String seaTunnelFilePath) {
...
+                }
+            }
+        } catch (Exception e){
+            log.error("delete no commit file failed", e);
```
Comment (FlechazoW) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/AbstractWriteStrategy.java]: Same as above.

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/AbstractWriteStrategy.java (hunk 3 above)
Comment (lightzhao) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/AbstractWriteStrategy.java]: complete modify.

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/AbstractWriteStrategy.java (hunk 4 above)
Comment (lightzhao) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/AbstractWriteStrategy.java]: complete modify.

**Comment by TyrantLucifer** [comment]
> > Can you provide test cases?
> 
> I have completed the test. When the checkpoint is not completed, I kill the flink job. The files in the temporary directory are deleted successfully.

Offer test case screenshot and write test cases, thanks.

BTW, if you choose offer test case screenshot, you need to provide all the steps to prove that this PR is valid and meaningful.

**Comment by lightzhao** [comment]
> > > Can you provide test cases?
> > 
> > 
> > I have completed the test. When the checkpoint is not completed, I kill the flink job. The files in the temporary directory are deleted successfully.
> 
> Offer test case screenshot and write test cases, thanks.
> 
> BTW, if you choose offer test case screenshot, you need to provide all the steps to prove that this PR is valid and meaningful.

I still need to continue the test. I found that the close method cannot be called after yarn kill. This may have something to do with the flink mechanism. So far, there is still a problem.

**Comment by lightzhao** [comment]
Flink's SinkWriter does not declare the close method, which cannot be called when yarn kill.

**Pull request closed without merging**